tests/test_tools_pr_comments.py

```diff
@@ -0,0 +1,86 @@
+# Tests for functions defined in 'tools/pr_comments.py' of the EESSI
+# build-and-deploy bot, see https://github.com/EESSI/eessi-bot-software-layer
+#
+# The bot helps with requests to add software installations to the
+# EESSI software layer, see https://github.com/EESSI/software-layer
+#
+# author: Thomas Roeblitz (@trz42)
+#
+# license: GPLv2
+#
+
+# Standard library imports
+
+# Third party imports (anything installed into the local Python environment)
+import pytest
+from unittest.mock import patch
+
+# Local application imports (anything from EESSI/eessi-bot-software-layer)
+from tools.pr_comments import get_comment, get_submitted_job_comment
+
+
+class MockIssueComment:
+    def __init__(self, body):
+        self.body = body
```

Comment:
Maybe use `namedtuple` for this (3 lines -> 1 line for `MockIssueComment` definition)

```
>>> Comment = namedtuple('Comment', ('body'))
>>> expected = Comment(body="foo")
>>> expected.body
'foo'
```

Reply:
That could work.

Would the `namedtuple` also allow to have method? That might be useful for testing `tools.pr_comments.update_comment`. The `.edit` function in there is a method of an `IssueComment`, so just adding it to `MockIssueComment` could be convenient. Alternatively, one could mock `IssueComment.edit` in the same way `*.PullRequest.get_issue_comments` is mocked.

Reply:
No, with `namedtuple` you can't mock methods, then you do need a mocking class like here.
That's a good reason to stick to the `class` approach like you're doing here.